src/libraries/Microsoft.PowerFx.Core/Public/Values/StringValue.cs

```diff
@@ -13,7 +13,7 @@ public class StringValue : PrimitiveValue<string>
         internal StringValue(IRContext irContext, string value)
             : base(irContext, value)
         {
-            Contract.Assert(IRContext.ResultType == FormulaType.String);
+            Contract.Assert(IRContext.ResultType == FormulaType.String || IRContext.ResultType == FormulaType.Hyperlink);
```

Comment:
This could be an issue - may break other invariants. I'm not sure. The alternative is a new HyperlinkValue : FormulaValue.